sdks/java/extensions/sql/zetasql/build.gradle

```diff
@@ -31,23 +32,29 @@ def zetasql_version = "2020.10.1"
 
 dependencies {
   compile enforcedPlatform(library.java.google_cloud_platform_libraries_bom)
-  compile project(":sdks:java:core")
+  compile project(path: ":sdks:java:core", configuration: "shadow")
   compile project(":sdks:java:extensions:sql")
+  compile project(":sdks:java:extensions:sql:udf")
   compile library.java.vendored_calcite_1_20_0
   compile library.java.guava
   compile library.java.grpc_api
+  compile library.java.joda_time
   compile library.java.protobuf_java
   compile library.java.protobuf_java_util
-  compile "com.google.api.grpc:proto-google-common-protos:1.12.0" // Interfaces with ZetaSQL use this
-  compile "com.google.api.grpc:grpc-google-common-protos:1.12.0" // Interfaces with ZetaSQL use this
-  compile "com.google.zetasql:zetasql-jni-channel:$zetasql_version"
+  permitUnusedDeclared library.java.protobuf_java_util
+  compile library.java.slf4j_api
+  compile library.java.vendored_guava_26_0_jre
+  compile library.java.proto_google_common_protos // Interfaces with ZetaSQL use this
+  permitUnusedDeclared "com.google.api.grpc:proto-google-common-protos:2.0.1" // BEAM-11761
```

Comment:
This was added twice.

Also, if you're using the `library.java` version in the `compile` statement, you should also use it in the corresponding `permitUnusedDeclared`.

Reply:
I couldn't get permitUnusedDeclared to work with the library.java variable when the version is coming from the bom. It seems permitUnusedDeclared sees a blank version and fails.

Reply:
@ibzib Regarding your comment about this being added twice, they are actually two different dependencies:
"com.google.api.grpc:proto-google-common-protos:2.0.1"
"com.google.api.grpc:grpc-google-common-protos:2.0.1"

It's possible both aren't really needed, but I think resolving that is probably a separate issue.

Reply:
> Regarding your comment about this being added twice, they are actually two different dependencies:

Oops, you're right. I didn't read carefully.

> I couldn't get permitUnusedDeclared to work with the library.java variable when the version is coming from the bom. It seems permitUnusedDeclared sees a blank version and fails.

We should be able to work around this by resolving the bom versions before running analyzeClassesDependencies. But I'm not sure exactly how. Filed BEAM-11823 to track.

Reply:
@shehzaadn-vd we can fix the issue with blank versions by using `enforcedPlatform`: https://github.com/ibzib/beam/commit/ef122510cdc719aa00badb78f0f6c4b30ef10b8b